src/Microsoft.DocAsCode.Build.Engine/HostServiceCreatorWithIncremental.cs

```diff
@@ -60,29 +60,37 @@ public override bool CanProcessorIncremental(IDocumentProcessor processor)
-                    if (IncrementalContext.ChangeDict.TryGetValue(fileKey, out ChangeKindWithDependency ck))
-                    {
-                        Logger.LogDiagnostic($"Processor {processor.Name}, File {file.FullPath}, ChangeType {ck}.");
-                        if (ck == ChangeKindWithDependency.Deleted)
-                        {
-                            return null;
-                        }
-                        if (ck == ChangeKindWithDependency.None)
-                        {
-                            Logger.LogDiagnostic($"Processor {processor.Name}, File {file.FullPath}: Check incremental...");
-                            if (processor.BuildSteps.Cast<ISupportIncrementalBuildStep>().All(step => step.CanIncrementalBuild(file)))
-                            {
-                                Logger.LogDiagnostic($"Processor {processor.Name}, File {file.FullPath}: Skip build by incremental.");
-                                return null;
-                            }
-                            Logger.LogDiagnostic($"Processor {processor.Name}, File {file.FullPath}: Incremental not available.");
-                        }
+            {
+                // isDependency always is false for full build
+                bool isDependency = false;
+                if (CanProcessorIncremental(processor))
+                {
+                    string fileKey = ((RelativePath)file.File).GetPathFromWorkingFolder().ToString();
+                    if (IncrementalContext.ChangeDict.TryGetValue(fileKey, out ChangeKindWithDependency ck))
+                    {
+                        isDependency = ck == ChangeKindWithDependency.DependencyUpdated;
+                        Logger.LogDiagnostic($"Processor {processor.Name}, File {file.FullPath}, ChangeType {ck}.");
+                        if (ck == ChangeKindWithDependency.Deleted)
+                        {
+                            return null;
+                        }
+                        if (ck == ChangeKindWithDependency.None)
+                        {
+                            Logger.LogDiagnostic($"Processor {processor.Name}, File {file.FullPath}: Check incremental...");
+                            if (processor.BuildSteps.Cast<ISupportIncrementalBuildStep>().All(step => step.CanIncrementalBuild(file)))
+                            {
+                                Logger.LogDiagnostic($"Processor {processor.Name}, File {file.FullPath}: Skip build by incremental.");
+                                return null;
+                            }
+                            Logger.LogDiagnostic($"Processor {processor.Name}, File {file.FullPath}: Incremental not available.");
+                        }
                     }
                 }
+
+                var fmetadata = metadata == null ? new Dictionary<string, object>() : new Dictionary<string, object>(metadata);
```

Comment:
>fmetadata [](start = 20, length = 9)

this will be in final output, how about set into Properties for the FileModel in PostCreate

Reply:
or manifest property?